lib/committee/middleware/request_validation.rb

```diff
@@ -2,25 +2,33 @@ module Committee::Middleware
   class RequestValidation < Base
     def initialize(app, options={})
       super
-      @allow_form_params  = options.fetch(:allow_form_params, true)
-      @allow_query_params = options.fetch(:allow_query_params, true)
-      @check_content_type = options.fetch(:check_content_type, true)
-      @optimistic_json    = options.fetch(:optimistic_json, false)
-      @strict             = options[:strict]
+      @allow_form_params   = options.fetch(:allow_form_params, true)
+      @allow_query_params  = options.fetch(:allow_query_params, true)
+      @check_content_type  = options.fetch(:check_content_type, true)
+      @optimistic_json     = options.fetch(:optimistic_json, false)
+      @coerce_query_params = options.fetch(:coerce_query_params, false)
+      @strict              = options[:strict]
 
       # deprecated
       @allow_extra = options[:allow_extra]
     end
 
     def handle(request)
-      request.env[@params_key] = Committee::RequestUnpacker.new(
-        request,
-        allow_form_params:  @allow_form_params,
-        allow_query_params: @allow_query_params,
-        optimistic_json:    @optimistic_json
-      ).call
-
       if link = @router.find_request_link(request)
+        if @coerce_query_params && !request.GET.nil? && !link.schema.nil?
+          request.env["rack.request.query_hash"].merge!(
+            Committee::QueryHashCoercer.new(
+              request.GET,
+              link.schema
+            ).call
+          )
+        end
+        request.env[@params_key] = Committee::RequestUnpacker.new(
```

Comment:
So one "feature" of Committee is that an unpacked set of parameters is still available in the request's env even if no link was found, and I think we need to keep this behavior. You might have to retool so that:

1. Look for link and run query coercer if appropriate.
2. Run request unpacker.
3. Re-use link variable from (1) and run the validator.